app/src/main/java/com/github/steroidteam/todolist/view/MapsActivity.java

```diff
@@ -0,0 +1,261 @@
+    private static final String KEY_CAMERA_POSITION = "camera_position";
+    private static final String KEY_LOCATION = "location";
+
+    private SearchView searchView;
+    private Marker marker;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Retrieve location and camera position from saved instance state.
+        if (savedInstanceState != null) {
+            lastKnownLocation = savedInstanceState.getParcelable(KEY_LOCATION);
+            cameraPosition = savedInstanceState.getParcelable(KEY_CAMERA_POSITION);
+        }
+
+        // Retrieve the content view that renders the map.
+        setContentView(R.layout.activity_maps);
+
+        searchView = findViewById(R.id.sv_location);
+
+        searchView.setOnQueryTextListener(createOnQueryTextListener());
+        // Construct a FusedLocationProviderClient.
+        fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this);
+
+        // Get the SupportMapFragment and request notification when the map is ready to be used.
+        SupportMapFragment mapFragment =
+                (SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.map);
+        mapFragment.getMapAsync(this::onMapReady);
+    }
+
+    private SearchView.OnQueryTextListener createOnQueryTextListener() {
+        return new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String s) {
+                String location = searchView.getQuery().toString();
+                List<Address> addressList = null;
+
+                if (location != null || !location.equals("")) {
+                    markerSearchLocation(location, addressList);
+                    searchView.setQuery("", false);
+                }
+                return false;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String s) {
+                return false;
+            }
+        };
+    }
+
+    private void markerSearchLocation(String location, List<Address> addressList) {
+        Geocoder geocoder = new Geocoder(MapsActivity.this);
+        try {
+            addressList = geocoder.getFromLocationName(location, 1);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        Address address = addressList.get(0);
+        LatLng latLng = new LatLng(address.getLatitude(), address.getLongitude());
+        String locationWithUppercase =
+                location.substring(0, 1).toUpperCase() + location.substring(1);
+        if (marker != null) {
+            marker.setPosition(latLng);
+            marker.setTitle(locationWithUppercase);
+        } else {
+            marker =
+                    map.addMarker(
+                            new MarkerOptions().position(latLng).title(locationWithUppercase));
+        }
+        map.animateCamera(CameraUpdateFactory.newLatLngZoom(latLng, 10));
```

Comment:
It will be great to have a private final attribute for the int 10, as you did for `DEFAULT_ZOOM `

Reply:
That's true I forgot to do it !